pkg/model/crd.go

```diff
@@ -1356,6 +1372,197 @@ func (r *CRD) GoCodeSetOutput(
 	return out
 }
 
+
+// Generates code string for custom update method per the generator config and diffReporter.
+// This is requires when Update method of the resource does not support fields that are supported by Create method of
+// same resource, and there exist specialized update methods on the resource api.
+//
+// For example, if generator.yaml contains following:
+//
+//	resources:
+//	  ReplicationGroup:
+//		custom_update_operations:
+//		  UpdateShardConfiguration:
+//			fields:
+//			  - NumNodeGroups
+//		  UpdateReplicaCount:
+//			fields:
+//			  - ReplicasPerNodeGroup
+//
+// Then invoking this method as {{ GoCodeCustomUpdate .CRD "ctx" "r" "diffReporter" 1 }} from template
+// should return:
+//		var ReplicationGroup *svcsdk.ReplicationGroup = nil
+//		for _, diff := range diffReporter.Differences {
+//			switch diff.Path {
+//			case "Spec.ReplicasPerNodeGroup":
+//				outputShapeName, respErr := rm.UpdateReplicaCount(ctx, r, diffReporter)
+//				if respErr != nil {
+//					return nil, respErr
+//				}
+//				ReplicationGroup = outputShapeName
+//			case "Spec.NumNodeGroups":
+//				outputShapeName, respErr := rm.UpdateShardConfiguration(ctx, r, diffReporter)
+//				if respErr != nil {
+//					return nil, respErr
+//				}
+//				ReplicationGroup = outputShapeName
+//			}
+//			if ReplicationGroup != nil {
```

Comment:
What happens if there are mutations to **both** the `Spec.ReplicasPerNodeGroup` **and** `Spec.NumNodeGroups` fields? In this code, only one of those calls to the custom callbacks will be triggered due to this line and `break`, no?

Reply:
Updated the code to `return` after invoking custom update method. Next reconciler invocation would lead to next custom update method call (if resource is in modifiable state).